torchvision/transforms/functional_tensor.py

```diff
@@ -226,8 +226,8 @@ def adjust_gamma(img: Tensor, gamma: float, gain: float = 1) -> Tensor:
 
     result = (gain * result ** gamma).clamp(0, 1)
 
-    result = convert_image_dtype(result, dtype)
-    result = result.to(dtype)
+    if result.dtype != dtype:
```

Comment:
@datumbox yes. But that line only checks if the image is already in that proposed dtype format. 
Since we are modifying the code for adjust_gamma. It needs the tensor to be in float dtype. So, if the image is of uint8 format,  it will change into float32 type.
Line no. 399 and 400 
https://github.com/pytorch/vision/blob/9e71fdafd871e3de9e72a6022291b49100945e29/torchvision/transforms/functional_tensor.py#L397-L406

But the method(adjust_gamma) will not return the image with its same original dtype format, unless later changed(currently inconsistent casting) . So, I think that's why we need a check later in the method.